### ISSUE TITLE
Character level beam search

### ISSUE THREAD

**Issue by skalinin** [pytorch/audio#2882]
### 🚀 The feature

Use ctc_decoder with character level KenLM

### Motivation, pitch

is there a way to use ctc_decoder with character level KenLM?

I have a text recognition model for reading text from images with one word only.

So I created kenlm using character n-grams (I separated chars in words using spaces, [for example](https://github.com/kpu/kenlm/issues/302))

And my kenlm .arpa file looks like this:
```
-0.12502529 <s> t h i n k
-0.33928923 s s w e r k
-0.16396953 m e w o r k
-0.09149383 e t w o r k
...
```

Without this LM the ctc_decoder works fine and decodes a word from the predicted logits (like word "laptop")
but If I enable my language model, the decoder will return only one character instead of a word (like " a ")

Can you please tell what I need to do to enable character level beam search? On this feature is not supported yet?

### Alternatives

The same .arpa-file works fine with [parlance/ctcdecode](https://github.com/parlance/ctcdecode)

**Comment by skalinin** [comment]
@carolineechen thanks for the answer!

By now I am using lexicon-free ctc_decoder.

I'll try to rebuild my kenlm model by adding a special token between words.

But as I am getting now, torchaudio beam search actually a [word beam search](https://towardsdatascience.com/word-beam-search-a-ctc-decoding-algorithm-b051d28f3d2e).
So I'll try also to add a lexicon to my decoding stage and make a word based kenlm.
